docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB.NET), which you sprinkle over otherwise normal methods. 
+
+A few important things to know before continuing:
+
+* ``Task<T>`` and ``Task`` are used to represent the return types for async code.  These are abstractions that can be thought of as "something that needs to get done and could be long-running".  `More on Task and Task<T>`_.
```

Comment:
I'd describe this slightly differently. I'd start with describing what Task represents and then state that it's the return type of async methods. One follows the other. There is no affinity to Task and long-running code. When I read long-running code, I think CPU-bound. A big scenario of async is I/O-bound code.